### PR TITLE
feat: add support for secant functionality `math/base/special/sec`

### LINKED ISSUE #225: [RFC]: add support for computing the secant (sec)
## Checklist

> Please ensure the following tasks are completed before submitting a feature request.

-   [x] Read and understood the [Code of Conduct][code-of-conduct].
-   [x] Searched for existing issues and pull requests.
-   [x] The issue name begins with `RFC:`.

## Description

> Description of the feature request.

This RFC proposes adding support for computing the secant. The initial implementation is straightforward:

```javascript
return 1.0 / cos( x );
```

Package: `@stdlib/math/base/special/sec`
Alias: `base.sec`

## Related Issues

> Does this feature request have any related issues?

No.

## Questions

> Any questions for reviewers?

No.

## Other

> Any other information relevant to this feature request? This may include screenshots, references, sample output, and/or implementation notes.

Reference implementations:

-   [Julia](https://github.com/JuliaLang/julia/blob/dee7d77c199c4776cf4e06afb7ce920f09ad3c8c/base/special/trig.jl#L958)

[code-of-conduct]: https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md

### PR DESCRIPTION
**Pull request by AbhijitRaut04**
Resolves #225 

Add the secant functionality using the existing cosine and sine functions.

The PR fullfills the purpose of calculating secant of a number.

This pull request:

-   is able to calculate secant of a number.

This pull request:

-   resolves #225 

## Questions

What modifications should I make for this PR ?

## Checklist
-   [x] Read, understood, and followed the [contributing guidelines](https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md)

@stdlib-js/reviewers

@kgryte 
@Planeshifter 

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/math/base/special/sec/README.md [hunk 1]
```diff
@@ -0,0 +1,215 @@
+<!--
+
+@license Apache-2.0
+
+Copyright (c) 2018 The Stdlib Authors.
```
Comment (headlessNode) [inline lib/node_modules/@stdlib/math/base/special/sec/README.md]: The standard practice is to add the current year when creating new files or packages. However, if you're only modifying existing files, the original year of creation should remain unchanged.

Comment (AbhijitRaut04) [reply lib/node_modules/@stdlib/math/base/special/sec/README.md]: Ok

Comment (AbhijitRaut04) [reply lib/node_modules/@stdlib/math/base/special/sec/README.md]: I have updated license as per year 2024

**Comment by AbhijitRaut04** [comment]
@headlessNode I have updated the License.

**Comment by AbhijitRaut04** [comment]
@kgryte Are there any changes required?

File: lib/node_modules/@stdlib/math/base/special/sec/src/main.c [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*
+*
+* ## Notice
+*
+* The following copyright, license, and long comment were part of the original implementation available as part of [FreeBSD]{@link https://svnweb.freebsd.org/base/release/12.2.0/lib/msun/src/s_tan.c}. The implementation follows the original, but has been modified for JavaScript.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/sec/src/main.c]: This doesn't actually point to the original implementation. This link is for kernel tan, not sec.

**Comment by kgryte** [comment]
@AbhijitRaut04 I have no idea where you got the implementation. It looks like you copied from some other package. The implementation for this package should be trivial and is explicitly provided in https://github.com/stdlib-js/stdlib/issues/225.

As such, this package needs several changes.

**Comment by kgryte** [comment]
Furthermore, given that tests and benchmarks are failing, it seems evident that you have not setup your local dev environment for linting, running tests, and running benchmarks. Please consult our contributing guide and our development guide.

**Comment by AbhijitRaut04** [comment]
@kgryte I have copied some part of tan and cos to get reference for the valid way of contributing. But I will try to sort it out.

**Comment by AbhijitRaut04** [comment]
@Planeshifter Can you please tell me how can I run tests on local environment ?

**Comment by AbhijitRaut04** [comment]
@kgryte I am working on this issue I will raise a new PR.

**Pull request closed without merging**